connectors/sources/sharepoint_online.py

```diff
@@ -699,27 +699,44 @@ async def site_list_role_assignments(self, site_web_url, site_list_name):
         except NotFound:
             return
 
-    async def site_list_items(self, site_id, list_id):
-        select = "createdDateTime,id,lastModifiedDateTime,weburl,createdBy,lastModifiedBy,contentType"
-        expand = "fields($select=Title,Link,Attachments,LinkTitle,LinkFilename,Description,Conversation)"
+    async def site_list_item_has_unique_role_assignments(
+        self, site_web_url, site_list_name, list_item_id
+    ):
+        self._validate_sharepoint_rest_url(site_web_url)
 
-        async for page in self._graph_api_client.scroll(
-            f"{GRAPH_API_URL}/sites/{site_id}/lists/{list_id}/items?$select={select}&$expand={expand}"
-        ):
-            for site_list in page:
-                yield site_list
+        url = f"{site_web_url}/_api/lists/GetByTitle('{site_list_name}')/items({list_item_id})/HasUniqueRoleAssignments"
```

Comment:
Curious - is this call necessary?

I see that you fetch `roleassignments` below, what if `HasUniqueRoleAssignments` is False, will `roleassignments` still return something that will break the logic?

Reply:
This logic is based on the Tim's work - I mirrored existing logic for site pages and site lists. 

If `HasUniqueRoleAssignments` returns `False` means that there are no custom roles defined and we resort to using parent permissions for access control for a given doc as in #1264